src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -80,6 +78,17 @@ function AxisSystem(props: Props): JSX.Element {
     ordinateInfo.isIndexAxis
   );
 
+  const xTickFormat = getTickFormatter(
+    visibleDomains[0],
+    width,
+    abscissaInfo.scaleType
+  );
+  const yTickFormat = getTickFormatter(
+    visibleDomains[1],
+    height,
+    ordinateInfo.scaleType
+  );
```

Comment:
`AxisSystem` is becoming a bit too complex for my taste, especially considering that the signature of `getTickFormatter` is very similar to that of `getTickProps`. There's clearly a need for a refactor, but I'll work on it if/once this is merged.

Reply:
Related to that, I find that `getTickProps` is a bad function in itself as the return value changes completely depending on a flag.

Reply:
Hmm can't think of a refactor that'd be worthwhile. If I simplify things somewhere, I complexify them somewhere else... My initial idea was to have something like this to replace lines 61 to 90:

```tsx
const [xAxisProps, xGridProps] = getAxisProps(visibleDomain[0], width, abscissaInfo);
const [yAxisProps, yGridProps] = getAxisProps(visibleDomain[1], height, ordinateInfo);
// ... where `<x|y>AxisProps` includes `scale`, `tickFormat` and (`numTicks` or `tickValues`)
// ... and `<x|y>GridProps` includes `scale` and (`numTicks` or `tickValues`)

// ... which I could all destructure easily in the JSX
<AxisBottom
  {...xAxisProps}
  {...AXIS_PROPS}
/>
```

But the range of the x-axis scale is `[0, width]`, while the range of the y-axis scale is `[height, 0]`, so it's not as easy... Assuming I keep the `scale` computation in `AxisSystem`, I could probably get rid of `getTicksProp` and move the logic for this in `getAxisProp`, but I don't think I could get rid of `getTickFormatter`... So I wonder whether it's worth the extra level of abstraction...

Reply:
To be discussed elsewhere more in depth ^^